packages/example-application/src/react-hook-form.tsx

```diff
@@ -0,0 +1,217 @@
+/* eslint-disable react/prop-types */
+
+import React, { useState, useCallback } from 'react';
+import * as GovUK from 'govuk-react';
+import { Link } from 'react-router-dom';
+
+import { useForm } from 'react-hook-form';
+
+const DateField = ({ input: { onChange, onBlur, ...input }, ...props }) => {
+  const [value, setValue] = useState(input.value);
+  return (
+    <GovUK.DateField
+      {...props}
+      input={{
+        onChange: (newValue) => {
+          setValue({ ...value, ...newValue });
+          onChange({ target: { value: { ...value, ...newValue }, name: input.name } });
+        },
+        onBlur: (newValue) => onBlur({ target: { value, name: input.name } }),
+        ...input,
+      }}
+    />
+  );
+};
+
+const ReactHookForm = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, submitCount },
+  } = useForm();
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [data, setData] = useState(false);
```

Comment:
what does `data` do? the name implies an object or shape of data rather than `boolean`.
maybe `hasData` if you're checking for data? or set the initial state to `{}`

Reply:
https://github.com/govuk-react/govuk-react/pull/916/commits/b780a5c383edddd9fa6c1ace743b1d4cef23049d